packages/engine-core/src/binary/BinaryEngine.ts

```diff
@@ -960,23 +970,33 @@ You very likely have the wrong "binaryTarget" defined in the schema.prisma file.
     } catch (e: any) {
       logger('req - e', e)
 
-      await this.handleRequestError(e, numTry <= MAX_REQUEST_RETRIES)
+      const { error, shouldRetry } = await this.handleRequestError(e)
+
+      const isWrite = ['create', 'update', 'delete'].some((method) => clientMethod.includes(method))
```

Comment:
I have made a shared util for the iswrite check.

Reply:
Just moving this to a function is an improvement, but it still happens on two different levels of abstraction inconsistently. I'll fix this.

Reply:
Also I think it would be safer to flip the condition and make it `isRead` because if a new write operation is added we will probably forget to add it here at first. I'll do that too if I won't be able to use the document type instead.